### PR TITLE
refactor & make it compatible with Zygote

### PR DESCRIPTION
**Pull request by Roger-luo**
I removed dependency on IRTools, since it seems to be easier to implement this one with Cassette itself already. I'll merge this later and setup a CI since this package just started... if there's no comments. I'll maintain this package from now as well.

### CODE REVIEW

File: src/pass.jl [hunk 1]
```diff
@@ -1,42 +1,67 @@
...
+end
 
-for f in [:, getproperty, LinearAlgebra.mul!, Base.promote_op, size, Base.to_shape,
-          Broadcast.broadcasted, Broadcast.instantiate, Broadcast.preprocess,
-          Broadcast.combine_eltypes, copyto!, Broadcast.copyto_nonleaf!,
-          Broadcast.axes]
-  @eval @inline (::Buffer)(::typeof($f), a...) = $f(a...)
+@inline function Cassette.overdub(ctx::BuffCtx, ::Type{Array{T, N}}, ::UndefInitializer, d::Vararg{Int, N}) where {T, N}
+    isbitstype(T) || return Array{T, N}(undef, d)
+    T === UInt8 && return Array{T, N}(undef, d) # make Zygote stack happy
```
Comment (Roger-luo) [inline src/pass.jl:41]: This is one place I had to workaround @MikeInnes but I'm not sure where is allocating those pullback stacks?

Comment (Roger-luo) [reply src/pass.jl:41]: If there's a type for stacks, I guess I could just add that type's constructor to white list then I won't need this line.

**Comment by Roger-luo** [comment]
There are some Cassette performance issue as well, I'll investigate that in another PR.

**Comment by MikeInnes** [comment]
Do we need the dependency on Zygote here? I think it'd be better to stick to IRTools for simplicity here.

**Comment by Roger-luo** [comment]
I think those are necessary to prevent potential slow down (at least based on last time I checked). Yeah I can change it to IRTools later, just doing mid term recently... and do you have time to look into that IR transformation issue in IRTools? I think I'll meet the same thing here https://github.com/MikeInnes/IRTools.jl/issues/21

**Comment by Roger-luo** [comment]
I noticed you've updated the repo. I'll close this then.

**Pull request closed without merging**